Fix broken error ouput for django.setup() call.

whoodes opened the pull request:
When settings.py is broken, Django doesn't start.  When Django doesn't
start, relevant data is never written to our errors.log file.

This isn't desirable, so we instead catch any `NameError`s that may
arise and log them appropriately.

Closes: #7032

Tested by:
* set up a production server
* verified it was running normally
* Added the line: `'Foo': bar` within `DEFAULT_SETTINGS` in `settings.py`
* ran `supervisorctl restart all`
* received `internal server error`
* confirmed the line: `2019-06-19 20:29:25,535 ERROR zproject.wsgi name 'bar' is not defined` was in `var/log/zulip/errors.log`.

whoodes commented:
@timabbott I tested this out in a production environment and it looks to be doing the job. Looking forward to your feedback :)

timabbott commented on zproject/wsgi.py:
```diff
@@ -15,14 +15,25 @@
+try:
+    django.setup()  # We need to call setup to load applications.
+except NameError as e:
+    logger.error(e)
```
Can we put the block that configures the logger here, so it only runs in the exception case?  

Also, we should reraise the exception to exit, rather than continuing execution...

timabbott replied:
(Also, I think if we use `logger.exception`, the full traceback will go in the log, which is probably what we want?)

timabbott commented:
This looks like a reasonable approach; let's test the improvements I suggested above.

whoodes commented:
@timabbott FYI :) I moved `logging` to be imported dynamically.  I also updated the comment.  Here's an example with the full traceback (definitely what we wanted):
```
2019-06-20 18:19:20,643 ERROR zproject.wsgi name 'bar' is not defined
Traceback (most recent call last):
  File "./zproject/wsgi.py", line 31, in <module>
    django.setup()  # We need to call setup to load applications.
  File "/home/zulip/deployments/2019-06-19-20-12-57/zulip-py3-venv/lib/python3.5/site-packages/django/__init__.py", line 22, in setup
    configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
  File "/home/zulip/deployments/2019-06-19-20-12-57/zulip-py3-venv/lib/python3.5/site-packages/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/home/zulip/deployments/2019-06-19-20-12-57/zulip-py3-venv/lib/python3.5/site-packages/django/conf/__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "/home/zulip/deployments/2019-06-19-20-12-57/zulip-py3-venv/lib/python3.5/site-packages/django/conf/__init__.py", line 110, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/srv/zulip-venv-cache/e1eb6592bce842070114364c4b6387c63e4c43d9/zulip-py3-venv/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "./zproject/settings.py", line 132, in <module>
    'FOO': bar,
NameError: name 'bar' is not defined
```

timabbott commented:
Merged as 3c1982ecaba91c966babebc5ec90621db55dca3e, after tweaking the comments for clarity and making it catch all exceptions, not just NameError (there's other syntax issues one can create in `/etc/zulip/settings.py`, and we want all of them to be logged properly).

Thanks @whoodes!  This will result in a nontrivial reduction in folks being confused when they make mistakes setting up a Zulip production system.
